addons/mail/models/mail_link_preview.py

```diff
@@ -123,11 +112,3 @@ def _to_store(self, store: Store, /):
                 load=False,
             )[0]
             store.add(preview, data)
-
-    @api.autovacuum
-    def _gc_mail_link_preview(self):
```

Comment:
Shouldn't we keep a garbage collect of old urls (more like 6 months old, 1 year old, ...) ? Anyway considering current internet usage, I bet most URLs will be broken sooner or later, so maybe we could consider that keeping one year old preview is not really useful (could come with something that re-fetches the preview if somehow we open an old record and preview was garbage collect, lazy approach).

Reply:
On the other hand, if the link doesn't work anymore, the link preview might have saved valuable information about it.
Maybe this could be done at a later time? It's not going to be an issue soon (or ever?)

Reply:
Oh I was just asking... more an open question indeed. As this kind of model accumulates data forever, it is a good practice to wonder what we want to do with the data we keep in DB. It is always going to consume more and more space for content we gradually stop using. So does it make sense to keep it ? We keep business data as it is important for statistics, history, knowing what was done, replying potential customer requests, ... but "side" data like this one can probably be garbage collected at some point. I am always in favor of removing old dusty records but letting you decide for this one :) .

Reply:
It's always harder to properly clean tables that use intermediate tables. I think it will make sense later. Or not at all since people share a limited amount of "unique" URLs. 
Now it's true that it will accumulate old data that may become irrelevant with time. While some think it has no value (and is annoying) I think it's part of the message, like the attachments. Not the same value, but part of the message anyway.

Reply:
Makes sense, anyway if someday it becomes an issue we can effectively act on it.

I suppose someday we will have to fight to the real issue: somehow archive / remove / scrap messages :D . And then we will have to talk about attachments, links, reactions, ... any side record. But that time has not come yet :mage:  !